src/services/pallets/PalletsStorageService.ts

```diff
@@ -102,6 +142,39 @@ export class PalletsStorageService extends AbstractService {
 		};
 	}
 
+	/**
+	 * This will grab either V13 or V14 metadata for pallets. The reason being is v14 introduced
+	 * `StorageEntryTypeV14` which is different from `StorageEntryTypeV13` when it comes
+	 * to `asMap`. This will ultimately give different responses, and we want to make sure
+	 * we preserve the integrity of older blocks.
+	 *
+	 * @param hash BlockHash to query
+	 */
+	private adjustMetadata = async (
+		hash: BlockHash,
+		adjustMetadataV13Arg: boolean
+	): Promise<MetadataV13 | MetadataV14> => {
+		const [historicMetadata, blockHeader, { specName }] = await Promise.all([
+			await this.api.rpc.state.getMetadata(hash),
+			await this.api.rpc.chain.getHeader(hash),
+			this.api.rpc.state.getRuntimeVersion(),
+		]);
+
+		const blockNumber = blockHeader.number.toNumber();
+
+		let adjustedMetadata;
```

Comment:
I'd probably name this just "metadata", otherwise the reader assumes it is adjusted some way when in reality it is not (most of the time, hopefully).

Reply:
Ahh so there is a metadata already (one of the query args). I will call it `chosenMetadata` instead